sigchanyzer.go

```diff
@@ -48,17 +48,25 @@ func run(pass *analysis.Pass) (interface{}, error) {
 			return
 		}
 		var chanDecl *ast.CallExpr
+
+		// track whether call.Args[0] is a *ast.CallExpr. This is true when make(chan os.Signal) is passed directly to signal.Notify
+		wasCallExpr := false
+
 		switch arg := call.Args[0].(type) {
 		case *ast.Ident:
 			if decl, ok := findDecl(arg).(*ast.CallExpr); ok {
 				chanDecl = decl
+				wasCallExpr = false
 			}
 		case *ast.CallExpr:
 			chanDecl = arg
+			wasCallExpr = true
 		}
-		if chanDecl == nil || len(chanDecl.Args) != 1 {
+
+		if chanDecl == nil || len(chanDecl.Args) != 1 || wasCallExpr {
 			return
 		}
```

Comment:
What we could do is clearly separate these cases, so firstly
```go
                if chanDecl == nil || len(chanDecl.Args) != 1 {
                        return
                }
        
                // There is one exception. If we've got:
                // signal.Notify(make(chan os.Signal), signal)
                // The Go runtime will write to the channel with
                // a non-blocking send. See https://golang.org/issues/45043
                if nonBlockingChanSend {
                        return
                }
```

Reply:
@odeke-em This is not exactly one exception, though. Above code will effectively ignore any call, not just the `make(chan os.Signal)`.

Also, the `nonBlockingChanSend` isn't meaningful. It's safe not because that runtime send to the channel using blocking send, but because the channel isn't read anywhere.